meta-balena-common/recipes-devtools/dosfstools/dosfstools_%.bbappend

```diff
@@ -1,5 +1,13 @@
 FILESEXTRAPATHS_append := ":${THISDIR}/${PN}"
-SRC_URI_append = " file://0001-src-check.c-Fix-up-mtools-created-bad-dir-entries.patch"
+
+python() {
+    from distutils.version import StrictVersion
+    package_version = d.getVar('PV', True)
+    src_uri = d.getVar('SRC_URI', True)
+    if StrictVersion(package_version) < StrictVersion('4.2'): # patch upstreamed in v4.2
+        d.setVar('SRC_URI',
+                 '{} file://0001-src-check.c-Fix-up-mtools-created-bad-dir-entries.patch'.format(src_uri))
+}
 
```

Comment:
@jakogut it's unconventional to use an anonymous python function to apply patches and it adds maintenance overhead to remove it when no longer required.
Just add a `dosfstools_4.1.bbappend` that applies the patch. The whole bbappend will then be dropped when updating Yocto versions.